sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c

```diff
@@ -309,9 +309,7 @@ static gnrc_rpl_parent_t *_gnrc_rpl_find_preferred_parent(gnrc_rpl_dodag_t *doda
         return NULL;
     }
 
-    LL_FOREACH(dodag->parents, elt) {
-        new_best = dodag->instance->of->which_parent(new_best, elt);
-    }
+    LL_SORT(dodag->parents, dodag->instance->of->which_parent);
```

Comment:
I would not advise to update this every time the path cost of a parent updates, that could be with every transmitted packet (or other event on which the path cost depends).

Recalculating this based on a timer or an event with some build in hysterisis (only update when the path cost changed by some amount) might be enough to lessen the burden of this.